modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/SynonymTokenFilterFactory.java

```diff
@@ -62,8 +76,10 @@ public class SynonymTokenFilterFactory extends AbstractTokenFilterFactory {
                     + "Instead, insert a lowercase filter in the filter chain before the synonym_graph filter."
             );
         }
+
+        this.synonymsSource = parseSettingsForSynonymsSource(settings);
```

Comment:
The synonyms source is now determined in the constructor so that all of the token filter's instance variables can remain `final`